apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -690,6 +691,16 @@ find_s(S, P, Dir) ->
         SubStr -> SubStr
     end.
 
+jq(FilterProgram, JSONBin)
+  when is_binary(FilterProgram), is_binary(JSONBin) ->
+    case jq:parse(FilterProgram, JSONBin) of
+        {ok, Result} -> erlang:iolist_to_binary(Result);
+        {error, ErrorReason} -> erlang:error(ErrorReason)
```

Comment:
Unless we need to log the stacktrace, do not raise error exception.
The rule engine entrypoint call catches all exceptions and log stacktrace for error but not throw.
```suggestion
        {error, ErrorReason} -> erlang:throw({jq_exception, ErrorReason})
```

Reply:
Thanks!